src/main/kotlin/carracing/game/domain/data/Car.kt

```diff
@@ -2,13 +2,45 @@ package carracing.game.domain.data
 
 private const val INITIAL_POSITION = 0
 private const val MIN_NUMBER_TO_MOVE = 4
+const val NAME_MAX_LENGTH = 5
 
-data class Car(
-    var position: Int = INITIAL_POSITION,
+class Car(
+    position: Int = INITIAL_POSITION,
+    name: String,
+    private val generateMoveNumber: () -> Int,
 ) {
-    fun move(number: Int) {
-        if (number >= MIN_NUMBER_TO_MOVE) {
+    var position: Int = position
+        private set
+
+    val name: String = validateName(name)
```

Comment:
I think it would be better to declare the properties in the primary constructor and perform validation in the initializer block. Also, when a function name starts with "validate", it's not immediately clear that it will return a value.